packages/grid/_modules_/grid/hooks/features/rowHeight/useGridRowHeight.ts

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { useGridLogger } from '../../utils/useGridLogger';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { useGridState } from '../../utils/useGridState';
+import { GridComponentProps } from '../../../GridComponentProps';
+import { useGridStateInit } from '../../utils/useGridStateInit';
+import { GridRowId } from '../../../models/gridRows';
+import { getCurrentPageRows } from '../../utils/useCurrentPageRows';
+import { useGridApiEventHandler } from '../../utils/useGridApiEventHandler';
+import { GridEvents } from '../../../models/events/gridEvents';
+import { GridRowHeightApi } from '../../../models/api/gridRowHeightApi';
+
+export const useGridRowHeight = (
+  apiRef: GridApiRef,
+  props: Pick<GridComponentProps, 'rowHeight' | 'getRowHeight' | 'pagination' | 'paginationMode'>,
+): void => {
+  const logger = useGridLogger(apiRef, 'useGridRowHeight');
+  const rowsHeightCollection = React.useRef(new Map<GridRowId, number>());
+
+  useGridStateInit(apiRef, (state) => ({
+    ...state,
+    currentRowsMeta: {
+      totalHeight: 0,
+      positions: [],
+    },
+  }));
+
+  const [gridState, setGridState, forceUpdate] = useGridState(apiRef);
+
+  const hydrateRowsMeta = React.useCallback(() => {
+    logger.debug(`Hydrating all visible rows`);
+    // console.log('Hydrating rows...')
+    // if (!gridState.pagination) {
+    //   return;
+    // }
+
+    const { rows } = getCurrentPageRows(apiRef.current.state, props);
```

Comment:
Here's the reason of why the skipped tests are failing. `props` changes every time and makes `hydrateRowsMeta` to be called. To fix, change to:

```ts
const { rows } = getCurrentPageRows(apiRef.current.state, {
  pagination: props.pagination,
  paginationMode: props.paginationMode,
});
```

Reply:
Thank you for checking the PR. I tried the suggestion but it's still failing the 2 skipped tests. The problem is coming from the `React.useEffect` used here but the other way is to subscribe/create events to react to the state changes listed in the `useEffect`.

Regarding not being a separate feature - true but then the way things are hooked up the variable row height depends on the pagination.